src/wallet/message_store.cpp

```diff
@@ -0,0 +1,1389 @@
+  member.index = 0;
+  member.auto_config_public_key = crypto::null_pkey;
+  member.auto_config_secret_key = crypto::null_skey;
+  member.auto_config_running = false;
+
+  for (size_t i = 0; i < m_coalition_size; ++i)
+  {
+    member.me = member.index == 0;    // Simple convention/automatism for now: The very first member is fixed as / must be "me"
+    m_members.push_back(member);
+    member.index++;
+  }
+
+  set_member(state, 0, own_label, own_transport_address, state.address);
+
+  m_nettype = state.nettype;
+  set_active(true);
+  m_filename = state.mms_file;
+  save(state);
+}
+
+void message_store::set_options(const boost::program_options::variables_map& vm)
+{
+  const options opts{};
+  auto bitmessage_address = command_line::get_arg(vm, opts.bitmessage_address);
+  auto bitmessage_login = command_line::get_arg(vm, opts.bitmessage_login);
+  set_options(bitmessage_address, bitmessage_login);
+}
+
+void message_store::set_options(const std::string &bitmessage_address, const std::string &bitmessage_login)
+{
+  m_transporter.set_options(bitmessage_address, bitmessage_login);
+}
+
+void message_store::set_member(const multisig_wallet_state &state,
+                               uint32_t index,
+                               const boost::optional<std::string> &label,
+                               const boost::optional<std::string> &transport_address,
+                               const boost::optional<cryptonote::account_public_address> monero_address)
+{
+  THROW_WALLET_EXCEPTION_IF(index >= m_coalition_size, tools::error::wallet_internal_error, "Invalid member index " + index);
+  coalition_member &m = m_members[index];
+  if (label)
+  {
+    m.label = label.get();
+  }
+  if (transport_address)
+  {
+    m.transport_address = transport_address.get();
+  }
+  if (monero_address)
+  {
+    m.monero_address_known = true;
+    m.monero_address = monero_address.get();
+  }
+  // Save to minimize the chance to loose that info (at least while in beta)
+  save(state);
+}
+
+const coalition_member &message_store::get_member(uint32_t index) const
+{
+  return m_members[index];
+}
+
+bool message_store::member_config_complete() const
+{
+  for (size_t i = 0; i < m_members.size(); ++i) {
+    const coalition_member &m = m_members[i];
+    if (m.label.empty() || m.transport_address.empty() || !m.monero_address_known) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// Check if all members have a label set (as it's a requirement for starting auto-config
+// by the "manager")
+bool message_store::member_labels_complete() const
+{
+  for (size_t i = 0; i < m_members.size(); ++i) {
+    const coalition_member &m = m_members[i];
+    if (m.label.empty()) {
+      return false;
+    }
+  }
+  return true;
+}
+
+void message_store::get_member_config(std::string &member_config)
+{
+  std::stringstream oss;
+  boost::archive::portable_binary_oarchive ar(oss);
+  ar << m_members;
+  member_config = oss.str();
+}
```

Comment:
No, there should be a `try ... catch`, here and in a similar situation a little below, for auto config data instead of member config data. Added in both places.